misk-redis/src/test/kotlin/misk/redis/FakeRedisTest.kt

```diff
@@ -760,4 +761,96 @@ class FakeRedisTest {
       redis.lrange(sourceKey, 0, -1)
     )
   }
+
+  @Test fun lpushAndLpop() {
+    val droids = listOf("4-LOM", "BB-8", "BD-1", "C-3PO", "IG-11", "IG-88B", "K-2SO", "R2-D2")
+      .map { it.encodeUtf8() }
+    redis.lpush("droids", *droids.toTypedArray())
+
+    // Non-expired keys.
+    assertThat(redis.lpop("droids", 1)).containsExactly("R2-D2".encodeUtf8())
+    assertThat(redis.lpop("droids", 3)).containsExactlyElementsOf(
+      listOf("K-2SO", "IG-88B", "IG-11").map { it.encodeUtf8() })
+    assertThat(redis.lpop("droids", 99)).containsExactlyElementsOf(
+      listOf("C-3PO", "BD-1", "BB-8", "4-LOM").map { it.encodeUtf8() })
+    assertThat(redis.lpop("droids", 1)).isEmpty()
+
+    // Expired key.
+    redis.lpush("droids", *droids.toTypedArray())
+    redis.pExpireAt("droids", clock.instant().toEpochMilli())
```

Comment:
If we don't expire them we can insert the same twice right?

Reply:
I can add a test for it, if you want, but yes, Redis allows for that.

Reply:
actually I think there's sufficient coverage here for lists with duplicate elements and access before expiry.